### PR TITLE
Update NetworkPolicy code for v1 semantics

### PR DESCRIPTION
**Pull request by danwinship**
With NetworkPolicy v1, there is no longer a "DefaultDeny" annotation, and instead, pods are isolated if and only if there is a NetworkPolicy selecting them. This updates for that.

First, when parsing NetworkPolicies, we note which pod IPs they select. Then when updating OVS, we output the "accept" flows for all the policies at `priority=150`, while also checking if any policy selects the entire Namespace. If there *is not* a policy that selects the entire Namespace, then that means there are potentially un-isolated pods that are default-accept rather than default-deny, so we also output a `priority=100, actions=drop` flow for each IP mentioned in any of the "accept" flows, and a `priority=50` flow accepting all other traffic in the namespace. (If there *is* a policy that selects the entire Namespace, then any traffic not accepted by any NetworkPolicy is denied, so we don't need any other rules because the base `priority=0, actions=drop` rule will do the right thing.)

We may want to revisit this and optimize it further later. In particular, it's inefficient if you have a different NetworkPolicy for each pod in a Namespace (and no match-all policies, and no unmatched pods). We could fix that by having explicit "accept" rules for unmatched pods rather than explicit "deny" rules for matched pods. That would have required more rewriting though, and it only matters in the case where you don't have a "default deny" policy anyway.

(Note to self: needs release note)

@openshift/networking PTAL

### CODE REVIEW

File: pkg/sdn/plugin/networkpolicy.go [hunk 1]
```diff
@@ -206,14 +199,28 @@ func (np *networkPolicyPlugin) syncNamespace(npns *npNamespace) {
 	otx := np.node.oc.NewTransaction()
 	otx.DeleteFlows("table=80, reg1=%d", npns.vnid)
 	if inUse {
-		if npns.isolated {
+		defaultAllow := true
```
Comment (knobunc) [inline pkg/sdn/plugin/networkpolicy.go]: Can we get a comment on this block?  Your commit message has great information about how this works, so just plop it in here too?

**Comment by knobunc** [comment]
[test]

**Comment by danwinship** [comment]
pushed with comments

**Comment by danwinship** [comment]
flake #14496 [test]

**Comment by danwinship** [comment]
flake #14496 again [test]

File: pkg/sdn/plugin/networkpolicy.go [hunk 2]
```diff
@@ -206,14 +199,39 @@ func (np *networkPolicyPlugin) syncNamespace(npns *npNamespace) {
...
+			// before that we need rules to drop any remaining traffic for any pod
+			// IP that *is* selected by a policy.
+			selectedIPs := sets.NewString()
 			for _, npp := range npns.policies {
-				for _, flow := range npp.flows {
-					otx.AddFlow("table=80, priority=100, reg1=%d, %s actions=output:NXM_NX_REG2[]", npns.vnid, flow)
+				for _, ip := range npp.selectedIPs {
+					if !selectedIPs.Has(ip) {
+						selectedIPs.Insert(ip)
+						otx.AddFlow("table=80, priority=100, reg1=%d, ip, nw_dst=%s, actions=drop", ip)
```
Comment (dcbw) [inline pkg/sdn/plugin/networkpolicy.go]: The AddFlow()'s format string has one more format than the # of args given.  Not sure what the intention here is, to keep the VNID or remove it.

Comment (danwinship) [inline pkg/sdn/plugin/networkpolicy.go]: oops, fixed

**Comment by danwinship** [comment]
maybe flake #9999. [test]

File: pkg/sdn/plugin/networkpolicy.go (hunk 2 above)
Comment (danwinship) [inline pkg/sdn/plugin/networkpolicy.go]: FWIW, this resulted in errors being logged while running the tests but didn't cause any of the tests to fail, which I guess means we need more test cases...

**Comment by danwinship** [comment]
flake #14496 again [test]

**Comment by knobunc** [comment]
test] last flake https://github.com/openshift/origin/issues/14180, logs: https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/2128/

**Comment by danwinship** [comment]
flake #14597 [test]

**Comment by danwinship** [comment]
pretty awesome that we've failed tests 6 times on a PR that doesn't modify any code that is exercised by any test...

**Comment by danwinship** [comment]
deleted all test comments. I had originally hoped we could get in a passing test run to pre-validate our eventual merge, but that's not going to happen...

**Comment by knobunc** [comment]
[merge]

**Comment by knobunc** [comment]
@danwinship Does this need a release note?

**Comment by danwinship** [comment]
Yes, we need a general NetworkPolicy stuff release note. It's on my todo list.

**Pull request merged**